src/Symfony/Component/Console/Helper/ProgressBar.php

```diff
@@ -391,6 +406,7 @@ private function overwrite(string $message): void
         }
 
         $this->firstRun = false;
+        $this->lastWriteTime = microtime(true);
```

Comment:
This is the wrong place to update `lastWriteTime` since there is no check if `display()` has been called at all, making this entire change moot.

Also I think it would be nicer to move this entire logic to the `display` function and not mix responsibility of setting progress and deciding whether or not display the changed progress in one function. So the display function will be called every time `setProgress` is called and it can then decide if it actually wants to display something.

Reply:
Ok I will move it to display.

Rest: This is not a point of this PR. I am following current logic. You can try to do it in different PR.

Reply:
I'd say either move all of to display or none of it and set lastWriteTime in both cases where display is called. Moving only lastWriteTime to display would be a very nasty case of temporal coupling.

Reply:
Ok, no moving then.